src/xma/src/xmaapi/xmaencoder.cpp

```diff
@@ -341,16 +342,53 @@ xma_enc_session_destroy(XmaEncoderSession *session)
     /*
     delete (XmaHwSessionPrivate*)session->base.hw_session.private_do_not_use;
     */
+    XmaHwSessionPrivate *priv1 = (XmaHwSessionPrivate*)session->base.hw_session.private_do_not_use;
```

Comment:
as Vamshi already suggested please use common functions where ever possible to avoid code duplicacy